File: app/src/main/java/com/agotakiss/androidtest/data/store/GenreStore.java
package com.agotakiss.androidtest.data.store;


import com.agotakiss.androidtest.data.models.GenreDataModel;
import com.agotakiss.androidtest.domain.models.Genre;

import java.util.List;
import java.util.Map;

import io.reactivex.Completable;
import io.reactivex.Single;

public interface GenreStore {

    Single<Boolean> hasData();

    Single<Map<Integer, GenreDataModel>> getGenreMap();

    Completable saveGenres(Map<Integer, GenreDataModel> genreMap);
}

File: app/src/main/java/com/agotakiss/androidtest/data/store/GenreStore.kt
package com.agotakiss.androidtest.data.store


import com.agotakiss.androidtest.domain.models.Genre
import io.reactivex.Completable
import io.reactivex.Single

interface GenreStore {

    fun getGenreMap(): Single<Map<Int, Genre>>

    fun hasData(): Single<Boolean>

    fun saveGenres(genreMap: Map<Int, Genre>): Completable
}
